docs:#1218 update ABS feedback from Beta (#1223)
redpanda-data/documentation: modules/deploy/pages/deployment-option/self-hosted/manual/sizing.adoc

In general, choose the number of nodes based on the following criteria, and add nodes for fault tolerance. This ensures that the system can operate with full throughput in a degraded state.

* **Network bandwidth**: Total bandwidth must account for maximum simultaneous writes and reads, multiplied by the replication factor.

> * **Memory per core**: Allocate a minimum of 2 GB memory for each CPU core. More memory improves performance.

Reviewer: ```suggestion
* **Memory per core**: Allocate a minimum of 2 GB memory for each CPU core. Not enough memory might lower performance.
```

Reviewer 2: About whether more memory improves performance: it's a big "it depends". For some workloads it will help, others it won't matter.

Reviewer: yeah that's why I suggested the opposite. It's more guaranteed that low memory impacts performance. Even catastrophically depending on how low.

Author: I changed to "Additional memory could improve performance."

> * **Memory per core**: Allocate a minimum of 2 GB memory for each CPU core. More memory improves performance.

Author: ```suggestion
* **Memory per core**: Allocate a minimum of 2 GB memory for each CPU core. Additional memory could improve performance.
```

* **Memory per partition**: Allocate a minimum of 2 MB of memory for each topic partition replica. 
+
For example: If you have 10,000 total partitions with a replication factor of 3, you need at least 60 GB of memory across all brokers:
+
10,000 partitions × 3 replicas × 2 MB = 60,000 MB (60 GB)
* **Storage strategy**: Use Tiered Storage to unify historical and real-time data cost-effectively.

> * **Performance testing**: Run hardware and Redpanda benchmark tests to establish a performance baseline. Choose instance types that prioritize storage and network performance:

Reviewer: ```suggestion
* **Performance testing**: Run hardware and Redpanda benchmark tests to establish a performance baseline. Choose instance types that prioritize network and storage performance:
```

nit, changed order because network is often more important than storage. its also much more expensive.

Reviewer 2: This advice is probably out of date. CPU power/$ is usually the most important, then or tied with reasonable local SSD performance (storage). Network is usually fine and in any case there aren't many knobs to tweak there, you get what you get with the general purpose instance types.

Reviewer: that's good info. Thanks @travisdowns

+
--

> * **AWS**: Test with i3en (NVMe SSD), i4i (latest NVMe), or is4gen (Intel-based NVMe)
> * **Azure**: Test with Lsv2-series (high I/O performance)
> * **GCP**: Test with n2-standard instances with local SSD

Reviewer: ```suggestion
* **AWS**: Test with i3en (NVMe SSD), i4i (NVMe), or is4gen (Intel-based NVMe) instances, or other NVMe-backed types
* **Azure**: Test with Lsv2-series instances (high I/O performance) or other high-performance storage types
* **GCP**: Test with n2-standard instances with local SSD or other high-performance local storage types
```

suggesting this for future-proof, avoiding things like "latest"

> * **GCP**: Test with n2-standard instances with local SSD

Reviewer 2: These don't reflect our current guidelines nor our actual instance types used in cloud. I have an internal page here which should be close to the truth:

https://redpandadata.atlassian.net/wiki/spaces/CORE/pages/1298792513/Instance+selection

H/e, I guess I can file a separate issue for it...

Author: Thanks @travisdowns: please file a separate issue. As part of reviewing/updating all the content, we could also do a style clean-up to make the content easier to read.
